chain/default_syncer.go

```diff
@@ -255,6 +255,9 @@ func (syncer *DefaultSyncer) syncOne(ctx context.Context, parent, next types.Tip
 	}
 
 	if heavier {
+		if syncer.IsReorg(ancestors, next) {
```

Comment:
observation: this logic could be implemented as a thingy that consumes the new heaviest tipset channel, assuming it allows multiple consumers. look for ways to implement things outside of major abstractions.